src/main/java/fi/aalto/cs/apluscourses/intellij/utils/SettingsUtil.java

```diff
@@ -0,0 +1,98 @@
+import org.jetbrains.annotations.NotNull;
+
+public class SettingsUtil {
+
+  /**
+   * Downloads the course IDE settings ZIP file to a temporary file. Also adds IDEA startup actions
+   * that unzip the temporary file to the IDEA configuration path after which the temporary file is
+   * deleted. Therefore, the new IDE settings only take effect once IDEA is restarted and the
+   * temporary file must still exist at that point.
+   * @throws IOException                 If an IO error occurs (e.g. network issues).
+   * @throws UnexpectedResponseException If an unexpected response is received when downloading the
+   *                                     project settings. This usually indicates an error in the
+   *                                     course configuration.
+   */
+  public static void importIdeSettings(@NotNull Course course)
+      throws IOException, UnexpectedResponseException {
+    URL ideSettingsUrl = course.getResourceUrls().get("ideSettings");
+    if (ideSettingsUrl == null) {
+      return;
+    }
+
+    File file = FileUtilRt.createTempFile("course-ide-settings", ".zip");
+    CoursesClient.fetchZip(ideSettingsUrl, file);
+    String configPath = FileUtilRt.toSystemIndependentName(PathManager.getConfigPath());
+    StartupActionScriptManager.addActionCommands(
+        Arrays.asList(
+            new StartupActionScriptManager.UnzipCommand(file, new File(configPath)),
+            new StartupActionScriptManager.DeleteCommand(file)
+        )
+    );
+
+    UpdateSettings.getInstance().forceCheckForUpdateAfterRestart();
+  }
+
+  /**
+   * Downloads the course project settings ZIP file to a temporary file. After that the files from
+   * the .idea directory of the ZIP file are extracted to the .idea directory of the given project,
+   * after which the project is reloaded. If the course does not provide custom project settings,
+   * this method does nothing.
+   * @throws IOException                 If an IO error occurs (e.g. network issues).
+   * @throws UnexpectedResponseException If an unexpected response is received when downloading the
+   *                                     project settings. This usually indicates an error in the
+   *                                     course configuration.
+   */
+  public static void importProjectSettings(@NotNull Project project, @NotNull Course course)
+      throws IOException, UnexpectedResponseException {
+    URL settingsUrl = course.getResourceUrls().get("projectSettings");
+    if (settingsUrl == null) {
+      return;
+    }
+
+    Path settingsPath = Paths.get(project.getBasePath(), Project.DIRECTORY_STORE_FOLDER);
+
+    File settingsZip = FileUtilRt.createTempFile(project.getName() + "-settings", ".zip");
+    CoursesClient.fetchZip(settingsUrl, settingsZip);
+    ZipFile zipFile = new ZipFile(settingsZip);
+    List<String> fileNames = zipFile
+        .getFileHeaders()
+        .stream()
+        .filter(file -> !file.isDirectory())
```

Comment:
should here be some extra filtering step for checking the `fileHeaders` against `null` or something like that?